app/models/queries/projects/filters/available_project_attributes_filter.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+#
+
+class Queries::Projects::Filters::AvailableProjectAttributesFilter < Queries::Projects::Filters::ProjectFilter
+  def self.key
+    :available_project_attributes
+  end
+
+  def type
+    :list_contains
+  end
+
+  def allowed_values
+    @allowed_values ||= ProjectCustomFieldProjectMapping.pluck(:custom_field_id).map { |id| [id, id.to_s] }
```

Comment:
The `allowed_values` method for a very long time consisted of a somewhat useless second value instead of only the id. That was because the API did not make use of the second value. But with the move to Hotwire, I recon that this is about to change. The second value is the label to be displayed. So if ever the filter is to be used in the UI, it should receive a human readable label, e.g. `ProjectCustomFieldProjectMapping.includes(:project_custom_field).pluck(:custom_field_id, :name)`

Reply:
Thanks! Turns out it's the other way round: the first value is the label to be displayed?

https://github.com/opf/openproject/blob/dev/app/models/queries/filters/strategies/list.rb#L54-L56

Reply:
Ah, thanks for the clarification.